### ISSUE TITLE
Only depend on coq-core in CI

### ISSUE THREAD

**Issue by rgrinberg** [ocaml/dune#4307]
Since coq is now split from its standard library (thanks @ejgallego!), we no longer need to install the full package to test our integration with the compiler. We can make this change once coq-core makes it to opam

**Comment by ejgallego** [comment]
Indeed, the test suite should work, main thing to do tho is that as of today, `coqc` will fail if the stdlib is not present and the option `-boot` is not passed, this is a bit annoying for testing. Actually, it could well be that people finally split the heaviest parts of the coq-stdlib package into a separate one, so coq-stdlib-prelude should be very light.

**Comment by ejgallego** [comment]
This should be now very easy after #12035 , only a couple of extraction tests do depend on the Stdlib.
